test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -283,3 +289,269 @@ def test_insufficient_sharding_dims(self):
+
+        if self.rank <= 1:
+            # Verify local shard.
+            local_shard = sharded_tensor.local_shards()[0]
+            self.assertEqual(torch.device(f'cuda:{self.rank}'), local_shard.tensor.device)
+            self.assertEqual((5, 5), local_shard.tensor.size())
+
+            # Verify local shard metadata.
+            self.assertEqual((self.rank * 5, 0), local_shard.metadata.shard_offsets)
+            self.assertEqual((5, 5), local_shard.metadata.shard_lengths)
+            self.assertEqual(f'rank:{self.rank}/cuda:{self.rank}', local_shard.metadata.placement)
+
+        # Verify global metadata.
+        sharding_metadata = sharded_tensor.sharding_metadata()
+        self.assertEqual(2, len(sharding_metadata))
+        for rank, shard_metadata in enumerate(sharding_metadata):
+            self.assertEqual((rank * 5, 0), shard_metadata.shard_offsets)
+            self.assertEqual((5, 5), shard_metadata.shard_lengths)
+            self.assertEqual(f'rank:{rank}/cuda:{rank}', shard_metadata.placement)
+
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_new_group(self):
+        self.init_pg()
+
+        spec = EnumerableShardingSpec([
+            ShardMetadata(
+                shard_offsets=[0, 0],
+                shard_lengths=[5, 5],
+                placement="rank:0/cuda:1",
+            ),
+            ShardMetadata(
+                shard_offsets=[5, 0],
+                shard_lengths=[5, 5],
+                placement="rank:2/cuda:3",
+            ),
+        ])
+
+        pg = dist.new_group(ranks=[1, 2, 3])
+
+        if self.rank >= 1:
+            sharded_tensor = _sharded_tensor.empty(spec, 10, 5, process_group=pg)
+            self.assertEqual((10, 5), sharded_tensor.size())
+            if self.rank == 1 or self.rank == 3:
+                # Verify local shard.
+                local_shard = sharded_tensor.local_shards()[0]
+                self.assertEqual(torch.device(f'cuda:{self.rank}'), local_shard.tensor.device)
+                self.assertEqual((5, 5), local_shard.tensor.size())
+
+                # Verify local shard metadata.
+                self.assertEqual((self.rank // 2 * 5, 0), local_shard.metadata.shard_offsets)
+                self.assertEqual((5, 5), local_shard.metadata.shard_lengths)
+                self.assertEqual(f'rank:{self.rank - 1}/cuda:{self.rank}', local_shard.metadata.placement)
+
+            # Verify global metadata.
+            sharding_metadata = sharded_tensor.sharding_metadata()
+            self.assertEqual(2, len(sharding_metadata))
+            for rank, shard_metadata in enumerate(sharding_metadata):
+                self.assertEqual((rank * 5, 0), shard_metadata.shard_offsets)
+                self.assertEqual((5, 5), shard_metadata.shard_lengths)
+                self.assertEqual(f'rank:{rank * 2}/cuda:{rank * 2 + 1}', shard_metadata.placement)
+
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_multiple_local_shards(self):
```

Comment:
Can you add a few more test cases to answer the following questions?
1) Do we allow overlap among different shards?
2) Do we allow the union of all the shards smaller than the input large tensor?

Reply:
I don't think we support any overlaps amongst any shards at the moment. All the shards should be mutually exclusive.

Reply:
Can we add a note at somewhere to declare this or shall we add a check to fail such invalid input?

Reply:
Makes sense I will update the docs to add this, regarding the check we have the check and a test for this as well: https://github.com/pytorch/pytorch/blob/master/test/distributed/_sharding_spec/test_sharding_spec.py#L158.